flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -376,4 +388,23 @@ private int generatePriority(ResourceProfile resourceProfile) {
 		}
 	}
 
+	/**
+	 * Get resource profile by given the container priority. It should always be got from the mapping
+	 * between resource profile and priority, otherwise it will throw the exception.
+	 *
+	 * @param priority The container priority used for distinguishing different resources.
+	 * @return The resource profile corresponding with the priority.
+	 * @throws Exception
+	 */
+	private ResourceProfile getResourceProfile(int priority) throws Exception{
+		if (resourcePriorities.containsValue(priority)) {
+			for (Map.Entry<ResourceProfile, Integer> entry : resourcePriorities.entrySet()) {
+				if (entry.getValue() == priority) {
+					return entry.getKey();
+				}
+			}
+		}
```

Comment:
Querying a map does seem a little trick, but the map is very small, for most flink job, it may only has several elements, as it matters only with the number of operators with different resources.